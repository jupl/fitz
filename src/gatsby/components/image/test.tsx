import {create} from 'react-test-renderer'

test('render with fixed Sharp image', async () => {
  const React = await import('react')
  const {GatsbyImage} = await import('.')
  const data = {
    childImageSharp: {
      fixed: {
        height: 100,
        src: '/some/url',
        srcSet: '/some/url 100w, /another/url 200w',
        width: 100,
      },
    },
    url: '/some/url',
  }
  const component = create(<GatsbyImage data={data} />)
  expect(component.toJSON()).toMatchSnapshot()
})

test('render with fluid Sharp image', async () => {
  const React = await import('react')
  const {GatsbyImage} = await import('.')
  const data = {
    childImageSharp: {
      fluid: {
        aspectRatio: 1,
        presentationWidth: 100,
        sizes: '(max-width: 50px) 50vw, 100px',
        src: '/some/url',
        srcSet: '/some/url 100w, /another/url 200w',
      },
    },
    url: '/some/url',
  }
  const component = create(<GatsbyImage data={data} />)
  expect(component.toJSON()).toMatchSnapshot()
})

test('render with no Sharp image', async () => {
  const React = await import('react')
  const {GatsbyImage} = await import('.')
  const data = {
    url: '/some/url',
  }
  const component = create(<GatsbyImage data={data} />)
  expect(component.toJSON()).toMatchSnapshot()
})
