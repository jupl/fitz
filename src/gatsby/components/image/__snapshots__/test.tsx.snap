// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with fixed Sharp image 1`] = `
<span
  className=" gatsby-image-wrapper"
  style={
    Object {
      "display": "inline-block",
      "height": 100,
      "overflow": "hidden",
      "position": "relative",
      "width": 100,
    }
  }
>
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "<picture><source srcset=\\"/some/url 100w, /another/url 200w\\" /><img loading=\\"lazy\\" width=\\"100\\" height=\\"100\\" srcset=\\"/some/url 100w, /another/url 200w\\" src=\\"/some/url\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
      }
    }
  />
</span>
`;

exports[`render with fluid Sharp image 1`] = `
<div
  className="gatsby-image-wrapper"
  style={
    Object {
      "maxHeight": null,
      "maxWidth": null,
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={true}
    style={
      Object {
        "paddingBottom": "100%",
        "width": "100%",
      }
    }
  />
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "<picture><source srcset=\\"/some/url 100w, /another/url 200w\\" sizes=\\"(max-width: 50px) 50vw, 100px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 50px) 50vw, 100px\\" srcset=\\"/some/url 100w, /another/url 200w\\" src=\\"/some/url\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
      }
    }
  />
</div>
`;

exports[`render with no Sharp image 1`] = `
<img
  className=""
  src="/some/url"
/>
`;
